python/src/equistore/operations/join.py

```diff
@@ -17,6 +17,40 @@ def join(tensormaps: List[TensorMap], axis: str):
     properties of the TensorMaps and if ``axis='samples'`` it will be the
     along the samples.
 
+    Possible clashes of the meta data like the property names
+    are resolved by the following strategy:
+
+    1. If the names (i.e property names) are the same and the
+       values (i.e property values)
+       are unique we keep the names and
+       join the values vertically. In example::
+
+        labels_1 = ["n"], [0, 2, 3]
+        labels_2 = ["n"], [1, 4, 5]
+
+      will lead to::
+
+        properties = ["n"], [0, 2, 3, 1, 4, 5]
+
+    2. If the names are the same but the values are not unique a new variable
+       "tensor" is added to the names. In example::
+
+            properties_1 = ["n"], [0, 2, 3]
+            properties_2 = ["n"], [0, 2]
+
+        will lead to::
+            properties = ["tensor", "n"], [[0, 0], [0, 2], [0, 3], [1, 0], [1, 2]]
+
+    3. If names of the Labels are different we change the names two to
+       ("tensor", "property"). In example::
+
+            properties_1 = ["a"], [0, 2, 3]
+            properties_2 = ["b", "c], [[0, 0], [1, 2]]
+
+        will lead to::
+
+            properties = ["tensor", "q"], [[0, 0], [0, 1], [0, 2], [1, 0], [1, 1]]
```

Comment:
I still have problems to understand this case. It would help me if it is explained in detail how the new labels correspond to the old ones. Let me give the second case as example
```
2. If the names are the same but the values are not unique a new variable
       "tensor" is added to the names. In example::
            properties_1 = ["n"], [0, 2, 3]
            properties_2 = ["n"], [0, 2]
        will lead to::
            joined_properties = ["tensor", "n"], [[0, 0], [0, 2], [0, 3], [1, 0], [1, 2]]

       in joined_properties the labels
       [0, 0], [0, 2], [0, 3] correspond to properties_1
       [1, 0], [1, 2] correspond to properties_2
```